### PR TITLE
feat: add support for Anthropic models (core only, no streaming)

### PR DESCRIPTION
**Pull request by fontanierh**
Adds support for Anthropic models (Claude) to dust core.
Streaming is supported by the Anthropic API but not yet implemented here.

I am doing a separate PR for the frontend part and I can work on adding streaming afterwards

### CODE REVIEW

File: front/pages/api/apps/[user]/[sId]/runs/index.ts [hunk 1]
```diff
@@ -62,17 +61,13 @@ async function handler(
     return;
   }
 
+  if (!auth.canReadApp(app)) {
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/pages/api/apps/[user]/[sId]/runs/index.ts:64]: ah I presume it's based off of your other PR?

[the remark this answers is not in the document]
Comment (fontanierh) [reply front/pages/api/apps/[user]/[sId]/runs/index.ts:64]: Yeah it should be but it is ! 
This PR is not ready though (draft mode)

**Comment by spolu** [comment]
Awesome first pass. I generally test the providers using the command line `dust` first to make sure everything works as expected before moving on to `front` :+1:

**Comment by fontanierh** [comment]
> I generally test the providers using the command line dust first to make sure everything works as expected before moving on to front 👍

Thank you ! will do

File: core/src/providers/anthropic.rs [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+    }
+
+    async fn test(&self) -> Result<()> {
+        if !utils::confirm(
+            "You are about to make a request for 1 token to `claude-instant-v1` on the Anthropic API.",
+        )? {
+            Err(anyhow!("User aborted Anthropic test."))?;
+        }
+
+        let mut llm = self.llm(String::from("claude-instant-v1"));
```
Comment (fontanierh) [inline core/src/providers/anthropic.rs:343]: Picked the smaller, faster model since I don't think we care

File: core/src/providers/anthropic.rs [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+        ))
+    }
+
+    async fn decode(&self, _tokens: Vec<usize>) -> Result<String> {
+        Err(anyhow!(
+            "Encode/Decode not implemented for provider `anthropic`"
+        ))
+    }
+
+    async fn chat(
```
Comment (spolu) [inline core/src/providers/anthropic.rs:246]: OOC, I haven't checked their API docs. Nothing like a chat-like format (given how Claude is geared towards chats?)

**Review by spolu — DISMISSED** [review]
LGTM

Left questions but feel free to merge as is

File: core/src/providers/anthropic.rs (hunk 3 above)
Comment (fontanierh) [reply core/src/providers/anthropic.rs:246]: No, they have only a single API endpoint as far as I can tell and it generates completions, not chat.
The Claude playground is a chat UI:
Screenshot 2023-04-17 at 15 00 56

They have "Edit Chat" at the top right, which reveals how it's done under the hood:
Screenshot 2023-04-17 at 15 02 43

File: core/src/providers/anthropic.rs [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+    pub stop_reason: StopReason,
+    pub stop: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct ErrorDetail {
+    pub loc: Vec<String>,
+    pub msg: String,
+    #[serde(rename = "type")]
+    pub type_: String,
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply core/src/providers/anthropic.rs]: `r#type` works great, moved to that 👍

**Review by spolu — APPROVED** [review]

**Pull request merged**